pkg/network/nmstate.go

```diff
@@ -23,7 +23,7 @@ const (
 
 // renderNMState generates the manifests of NMState handler
 func renderNMState(conf *cnao.NetworkAddonsConfigSpec, manifestDir string, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
-	if conf.NMState == nil {
+	if conf.NMState == nil || (clusterInfo.NmstateOperator && clusterInfo.NmstateCRExists) {
```

Comment:
can you move the `clusterInfo.NmstateOperator && clusterInfo.NmstateCRExists)` logic to a local function for better readability? give it a name that would better explain why CNAO no longer needs to render the manifests..

Reply:
[Done](https://github.com/kubevirt/cluster-network-addons-operator/compare/35f776d9be08fce4039c5feeb2a2cfa0f8faec9e..f099f4260b20dbac306d4461103bdfcc0b3328c7)
+
[renamed the function](https://github.com/kubevirt/cluster-network-addons-operator/compare/f099f4260b20dbac306d4461103bdfcc0b3328c7..9e14dfb171a051762742193fc844ca628a8fc76b)

Reply:
you added the change to the e2e commit which is irelevant. can you move it to the first commit?

Reply:
Right, thanks, moved to the first commit